### ISSUE TITLE
protocol loading screen has spurious renders

### ISSUE THREAD

**Issue by rebeccamadsen** [codaco/Network-Canvas#727]
This may just be solvable with a "close" button on the loading screen, but sometimes the loading screen gets stuck. I've seen it in electron when doing development and saving a change that makes the build error and then saving a change that doesn't error. But I've also seen this "stuck" state on electron and Android emulator when attempting to load a protocol with errors. Trying to load a protocol from the paired server stalls after Download and Import steps (Android):
```
Uncaught (in promise) FileError {code: 1}code: 1__proto__: Object
```

In Electron I get a dialog, but still get stuck:
```
index.js:2178 Error: ENOENT: no such file or directory, open 'C:\Users\Big Baloo\AppData\Roaming\Network Canvas\protocols\257e210a-e461-42d5-ba59-9c5136dee238\assets\click_the_thing.mp3'
__stack_frame_overlay_proxy_console__ @ index.js:2178
reducer @ errors.js:22
combination @ combineReducers.js:120
rootReducer @ rootReducer.js:40
(anonymous) @ autoRehydrate.js:30
dispatch @ createStore.js:165
(anonymous) @ index.js:104
(anonymous) @ createEpicMiddleware.js:59
(anonymous) @ index.js:11
(anonymous) @ middleware.js:13
dispatch @ applyMiddleware.js:35
./node_modules/rxjs/Subscriber.js.SafeSubscriber.__tryOrUnsub @ Subscriber.js:242
./node_modules/rxjs/Subscriber.js.SafeSubscriber.next @ Subscriber.js:189
./node_modules/rxjs/Subscriber.js.Subscriber._next @ Subscriber.js:129
./node_modules/rxjs/Subscriber.js.Subscriber.next @ Subscriber.js:93
./node_modules/rxjs/operators/switchMap.js.SwitchMapSubscriber.notifyNext @ switchMap.js:126
./node_modules/rxjs/InnerSubscriber.js.InnerSubscriber._next @ InnerSubscriber.js:23
./node_modules/rxjs/Subscriber.js.Subscriber.next @ Subscriber.js:93
./node_modules/rxjs/operators/mergeMap.js.MergeMapSubscriber.notifyNext @ mergeMap.js:145
./node_modules/rxjs/InnerSubscriber.js.InnerSubscriber._next @ InnerSubscriber.js:23
./node_modules/rxjs/Subscriber.js.Subscriber.next @ Subscriber.js:93
./node_modules/rxjs/operators/mergeMap.js.MergeMapSubscriber.notifyNext @ mergeMap.js:145
./node_modules/rxjs/InnerSubscriber.js.InnerSubscriber._next @ InnerSubscriber.js:23
./node_modules/rxjs/Subscriber.js.Subscriber.next @ Subscriber.js:93
./node_modules/rxjs/operators/switchMap.js.SwitchMapSubscriber.notifyNext @ switchMap.js:126
./node_modules/rxjs/InnerSubscriber.js.InnerSubscriber._next @ InnerSubscriber.js:23
./node_modules/rxjs/Subscriber.js.Subscriber.next @ Subscriber.js:93
./node_modules/rxjs/OuterSubscriber.js.OuterSubscriber.notifyNext @ OuterSubscriber.js:19
./node_modules/rxjs/InnerSubscriber.js.InnerSubscriber._next @ InnerSubscriber.js:23
./node_modules/rxjs/Subscriber.js.Subscriber.next @ Subscriber.js:93
subscribeToResult @ subscribeToResult.js:17
./node_modules/rxjs/operators/catchError.js.CatchSubscriber.error @ catchError.js:111
./node_modules/rxjs/Subscriber.js.Subscriber._error @ Subscriber.js:132
./node_modules/rxjs/Subscriber.js.Subscriber.error @ Subscriber.js:106
(anonymous) @ PromiseObservable.js:71
Promise rejected (async)
./node_modules/rxjs/observable/PromiseObservable.js.PromiseObservable._subscribe @ PromiseObservable.js:62
./node_modules/rxjs/Observable.js.Observable._trySubscribe @ Observable.js:172
./node_modules/rxjs/Observable.js.Observable.subscribe @ Observable.js:160
./node_modules/rxjs/operators/map.js.MapOperator.call @ map.js:56
./node_modules/rxjs/Observable.js.Observable.subscribe @ Observable.js:157
./node_modules/rxjs/operators/catchError.js.CatchOperator.call @ catchError.js:79
./node_modules/rxjs/Observable.js.Observable.subscribe @ Observable.js:157
subscribeToResult @ subscribeToResult.js:23
./node_modules/rxjs/operators/switchMap.js.SwitchMapSubscriber._innerSub @ switchMap.js:103
./node_modules/rxjs/operators/switchMap.js.SwitchMapSubscriber._next @ switchMap.js:96
./node_modules/rxjs/Subscriber.js.Subscriber.next @ Subscriber.js:93
./node_modules/rxjs/operators/filter.js.FilterSubscriber._next @ filter.js:89
./node_modules/rxjs/Subscriber.js.Subscriber.next @ Subscriber.js:93
./node_modules/rxjs/Subject.js.Subject.next @ Subject.js:55
(anonymous) @ createEpicMiddleware.js:60
(anonymous) @ index.js:11
(anonymous) @ middleware.js:13
dispatch @ applyMiddleware.js:35
./node_modules/rxjs/Subscriber.js.SafeSubscriber.__tryOrUnsub @ Subscriber.js:242
./node_modules/rxjs/Subscriber.js.SafeSubscriber.next @ Subscriber.js:189
./node_modules/rxjs/Subscriber.js.Subscriber._next @ Subscriber.js:129
./node_modules/rxjs/Subscriber.js.Subscriber.next @ Subscriber.js:93
./node_modules/rxjs/operators/switchMap.js.SwitchMapSubscriber.notifyNext @ switchMap.js:126
./node_modules/rxjs/InnerSubscriber.js.InnerSubscriber._next @ InnerSubscriber.js:23
./node_modules/rxjs/Subscriber.js.Subscriber.next @ Subscriber.js:93
./node_modules/rxjs/operators/mergeMap.js.MergeMapSubscriber.notifyNext @ mergeMap.js:145
./node_modules/rxjs/InnerSubscriber.js.InnerSubscriber._next @ InnerSubscriber.js:23
./node_modules/rxjs/Subscriber.js.Subscriber.next @ Subscriber.js:93
./node_modules/rxjs/operators/mergeMap.js.MergeMapSubscriber.notifyNext @ mergeMap.js:145
./node_modules/rxjs/InnerSubscriber.js.InnerSubscriber._next @ InnerSubscriber.js:23
./node_modules/rxjs/Subscriber.js.Subscriber.next @ Subscriber.js:93
./node_modules/rxjs/operators/switchMap.js.SwitchMapSubscriber.notifyNext @ switchMap.js:126
./node_modules/rxjs/InnerSubscriber.js.InnerSubscriber._next @ InnerSubscriber.js:23
./node_modules/rxjs/Subscriber.js.Subscriber.next @ Subscriber.js:93
./node_modules/rxjs/Subscriber.js.Subscriber._next @ Subscriber.js:129
./node_modules/rxjs/Subscriber.js.Subscriber.next @ Subscriber.js:93
./node_modules/rxjs/operators/map.js.MapSubscriber._next @ map.js:85
./node_modules/rxjs/Subscriber.js.Subscriber.next @ Subscriber.js:93
(anonymous) @ PromiseObservable.js:66
Promise resolved (async)
./node_modules/rxjs/observable/PromiseObservable.js.PromiseObservable._subscribe @ PromiseObservable.js:62
./node_modules/rxjs/Observable.js.Observable._trySubscribe @ Observable.js:172
./node_modules/rxjs/Observable.js.Observable.subscribe @ Observable.js:160
./node_modules/rxjs/operators/map.js.MapOperator.call @ map.js:56
./node_modules/rxjs/Observable.js.Observable.subscribe @ Observable.js:157
./node_modules/rxjs/operators/catchError.js.CatchOperator.call @ catchError.js:79
./node_modules/rxjs/Observable.js.Observable.subscribe @ Observable.js:157
subscribeToResult @ subscribeToResult.js:23
./node_modules/rxjs/operators/switchMap.js.SwitchMapSubscriber._innerSub @ switchMap.js:103
./node_modules/rxjs/operators/switchMap.js.SwitchMapSubscriber._next @ switchMap.js:96
./node_modules/rxjs/Subscriber.js.Subscriber.next @ Subscriber.js:93
./node_modules/rxjs/operators/filter.js.FilterSubscriber._next @ filter.js:89
./node_modules/rxjs/Subscriber.js.Subscriber.next @ Subscriber.js:93
./node_modules/rxjs/Subject.js.Subject.next @ Subject.js:55
(anonymous) @ createEpicMiddleware.js:60
(anonymous) @ index.js:11
(anonymous) @ middleware.js:13
(anonymous) @ bindActionCreators.js:3
(anonymous) @ ServerProtocols.js:51
Promise resolved (async)
ServerProtocols._this.handleSelectProtocol @ ServerProtocols.js:50
selectProtocol @ ServerProtocolList.js:33
onClick @ ProtocolCard.js:19
callCallback @ react-dom.development.js:145
invokeGuardedCallbackDev @ react-dom.development.js:195
invokeGuardedCallback @ react-dom.development.js:248
invokeGuardedCallbackAndCatchFirstError @ react-dom.development.js:262
executeDispatch @ react-dom.development.js:593
executeDispatchesInOrder @ react-dom.development.js:615
executeDispatchesAndRelease @ react-dom.development.js:713
executeDispatchesAndReleaseTopLevel @ react-dom.development.js:724
forEachAccumulated @ react-dom.development.js:694
runEventsInBatch @ react-dom.development.js:855
runExtractedEventsInBatch @ react-dom.development.js:864
handleTopLevel @ react-dom.development.js:4857
batchedUpdates$1 @ react-dom.development.js:17498
batchedUpdates @ react-dom.development.js:2189
dispatchEvent @ react-dom.development.js:4936
interactiveUpdates$1 @ react-dom.development.js:17553
interactiveUpdates @ react-dom.development.js:2208
dispatchInteractiveEvent @ react-dom.development.js:4913
index.js:2178 Warning: Each child in an array or iterator should have a unique "key" prop.

Check the render method of `Dialog`. It was passed a child from ErrorMessage. See https://fb.me/react-warning-keys for more information.
    in p (at ErrorMessage.js:17)
    in Dialog (at ErrorMessage.js:34)
    in ErrorMessage (created by Connect(ErrorMessage))
    in Connect(ErrorMessage) (at App.js:41)
    in div (at App.js:19)
    in App (created by Connect(App))
    in Connect(App) (created by Route)
    in Route (created by withRouter(Connect(App)))
    in withRouter(Connect(App)) (at src/index.js:16)
    in Router (created by ConnectedRouter)
    in ConnectedRouter (at src/index.js:15)
    in Provider (at src/index.js:14)
```

It's probably a bad protocol, but getting stuck is worse.

**Comment by jthrilly** [comment]
I experienced this for the first time today, also on an android device.

**Comment by bryfox** [comment]
See #754 for some more explanation of the issue.

I also notice that minimumTimeMet is enforced even if there's an immediate error when downloading a protocol. Example: import from an invalid URL. It's not a big deal since the error modal is on top of the loading screen, but it might be nice to consider this in the future.

**Comment by bryfox** [comment]
I updated title & labels since the immediate problems were fixed in #754

**Comment by jthrilly** [comment]
I'm closing this as the underlying components have substantially changes since this issue.
